include/barrett/config.h

```diff
@@ -0,0 +1,30 @@
+#ifndef __LIBBARRETT_CONFIG_H
```

Comment:
Why is this file necessary? I think CMake rewrites `include/barrett/config.h.in` to produce this file.

Reply:
Interestingly, after deleting the file and then re-running cmake, the file did not re-generate.

This is in contrast to `cmake/barrett-config.cmake`, which did re-generate.

So, for whatever reason, it appears to be necessary.

Reply:
The `config.h` file wouldn't be regenerated in the source directory. [This](https://github.com/personalrobotics/libbarrett/blob/upgrade/xenomai3/CMakeLists.txt#L206) shows that it should be written to `${PROJECT_BINARY_DIR}/include/barrett/config.h`. In contrast, `cmake/barrett-config.cmake.in` is written to `${PROJECT_SOURCE_DIR}/cmake/barrett-config.cmake` so it looks like it regenerates.

The question to me is whether this compiles with nothing at `${PROJECT_SOURCE_DIR}/include/barrett/config.h`. I think [it should](https://github.com/personalrobotics/libbarrett/blob/upgrade/xenomai3/CMakeLists.txt#L79), but it'd be great to confirm on `herb0`. If that's the case, then I think we can just delete it.

Reply:
config.h removed